thirdparty/glslang/glslang/Include/Common.h

```diff
@@ -50,7 +50,9 @@ std::string to_string(const T& val) {
 }
 #endif
 
-#if (defined(_MSC_VER) && _MSC_VER < 1900 /*vs2015*/) || defined MINGW_HAS_SECURE_API
+// -- GODOT start --
+#if (defined(_MSC_VER) && _MSC_VER < 1900 /*vs2015*/) /* || defined MINGW_HAS_SECURE_API*/
```

Comment:
What was the issue with `MINGW_HAS_SECURE_API`? Would it be worth reporting upstream?

Reply:
Build fails with the following error.
```
stdio_s.h:495:23: error: conflicting declaration of 'int sprintf_s(char*, size_t, const char*, ...)' with 'C' linkage
   _SECIMP int __cdecl sprintf_s(char *_DstBuf,size_t _DstSize,const char *_Format,...);
                       ^~~~~~~~~
In file included from thirdparty/glslang/glslang/MachineIndependent/../Include/intermediate.h:55,
                 from thirdparty/glslang/glslang/MachineIndependent/RemoveTree.cpp:37:
thirdparty/glslang/glslang/MachineIndependent/../Include/../Include/Common.h:58:22: note: previous declaration with 'C++' linkage
     #define snprintf sprintf_s
                      ^~~~~~~~~
```

Some sort of include order issue, this one probably need more investigation to make sure it's not related to the configuration of the macOS homebrew MinGW package configuration.

In the upstream it was changed back and forth few times, same error is mentioned in the upstream PR changing include order (https://github.com/KhronosGroup/glslang/pull/203/files#r56031993). bgfx have a similar fix (https://github.com/bkaradzic/bgfx/pull/1352) for example. Seems to be known problem that's coming back time to time.